src/infra/dataverse/sparql/sparqlGateway.ts

```diff
@@ -22,11 +22,11 @@ export const sparqlGateway: DataversePort = {
     offset: number,
     filters: RetrieveDataverseQueryFilters
   ): TE.TaskEither<Error, RetrieveDataverseResult> => {
-    const buildContainsExpression = (filter: DataverseElementType): string =>
-      `contains(str(?type), "${filter}" )`
+    const buildStrExpression = (filter: DataverseElementType): string =>
+      `str(?type) = str(core:${filter})`
```

Comment:
I would remove the operator `str` on both sides to make a plain URI comparison:

```suggestion
    const buildStrExpression = (filter: DataverseElementType): string =>
      `?type = core:${filter}`
```

Reply:
Yes, you're right, I thought it was better to compare literals over URIs but I was mistaken.